roxie/udplib/udptopo.cpp

```diff
@@ -184,18 +193,38 @@ CTopologyServer::CTopologyServer(const char *topologyInfo)
             }
             if (streq(role, "agent"))
             {
-                agents[channel].append(ep);
-                if (ep.equals(myAgentEP))
+                if (instance)
                 {
-                    mySubChannels[channel] = agents[channel].ordinality()-1;
-                    channels.push_back(channel);
-                    replicationLevels.push_back(repl);
+                    agents[channel].append(ep);
+                    if (ep.equals(myAgentEP))
+                    {
+                        mySubChannels[channel] = agents[channel].ordinality()-1;
+                        channels.push_back(channel);
+                        replicationLevels.push_back(repl);
+                    }
+                    agents[0].append(ep);
                 }
-                agents[0].append(ep);
             }
             else if (streq(role, "server"))
-                servers[ep.port].append(ep);
+            {
+                time_t oldInstance = old ? old->queryServerInstance(ep) : 0;
+                if (!instance || (oldInstance && oldInstance != instance))
+                {
+                    StringBuffer s;
+                    DBGLOG("Deleting pending data for server %s which has terminated or restarted", ep.getUrlStr(s).str());
+                    ROQ->abortPendingData(ep);
+                }
+                if (instance)
+                {
+                    servers[ep.port].append(ep);
+                    serverInstances[ep] = instance;
+                }
+                else
+                    serverInstances.erase(ep);
```

Comment:
I don't think this will be there anyway - because you are creating a new entry?

Reply:
I suspect you are right